### PR TITLE
Fix handling of retargeted fixed size fields

### PR DESCRIPTION
**Pull request by jbevain**
Hopefully should fix the following internal issue:

https://developercommunity.visualstudio.com/content/problem/1075667/cs8385-incorrectly-reported-through-intellisense.html

Fixes [AB#1141012](https://devdiv.visualstudio.com/0bdbc590-a062-4c3f-b0f6-9383f67865ee/_workitems/edit/1141012)

### CODE REVIEW

**Review by mavasani — APPROVED** [review]
Nice!

**Comment by mavasani** [comment]
@dotnet/roslyn-compiler for reviews.

**Review by cston — APPROVED** [review]

File: src/Compilers/CSharp/Test/Symbol/Symbols/Retargeting/RetargetingTests.cs [hunk 1]
```diff
@@ -589,6 +589,26 @@ public interface Test : short { }
...
+
+            var comp = CreateCompilation(source);
+            var sourceAssembly = (SourceAssemblySymbol)comp.Assembly;
+            var retargetingAssembly = new RetargetingAssemblySymbol(sourceAssembly, isLinked: false);
+            retargetingAssembly.SetCorLibrary(MissingCorLibrarySymbol.Instance); // Need to do this explicitly since our retargeting assembly wasn't constructed using the real mechanism.
+            var retargetingType = retargetingAssembly.GlobalNamespace.GetMember<NamedTypeSymbol>("Test");
+            var retargetingField = retargetingType.GetMember<RetargetingFieldSymbol>("buffer");
+            Assert.True(retargetingField.IsFixedSizeBuffer);
+        }
+
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Symbol/Symbols/Retargeting/RetargetingTests.cs:611]: Thanks for covering the change. However, this type of test for retargeting symbols has fallen in disfavor, I was told.

Here's a template for creating retargeting symbols: create two compilations with two different corlibs. Such test can be put in `FixedSizeBufferTests`.

```
            var source = @"
namespace System
{
    public struct ValueTuple<T1, T2>
    {
        public T1 Item1;
        public T2 Item2;
        public string field;
    }
}";
            var comp = CreateCompilation(source, targetFramework: TargetFramework.Mscorlib40, assemblyName: "customValueTuple");
            comp.VerifyDiagnostics();
           
            var comp3 = CreateCompilation("", references: new[] { comp.ToMetadataReference() }, targetFramework: TargetFramework.Mscorlib46);
            var retargetingValueTupleType = (NamedTypeSymbol)comp3.GlobalNamespace.GetMember<NamespaceSymbol>("System").GetMembers("ValueTuple").Single();
            Assert.IsType<RetargetingNamedTypeSymbol>(retargetingValueTupleType);
```

**Review by jcouv — COMMENTED** [review]
Done with review (iteration 2)

**Review by mavasani — APPROVED** [review]

**Review by cston — APPROVED** [review]

**Comment by mavasani** [comment]
@jbevain Seems like the added test is failing.

**Comment by jbevain** [comment]
@mavasani yep, looks like I need to configure the Compilation to allow usafe code.

**Review by jcouv — APPROVED** [review]
LGTM (iteration 3). Thanks much!

**Comment by jcouv** [comment]
Merged/squashed. Thanks @jbevain for the fix!

**Pull request merged**